src/main/java/com/palantir/common/streams/ConcurrentStreams.java

```diff
@@ -0,0 +1,120 @@
+package com.palantir.common.streams;
+
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.immutables.value.Value;
+
+public class ConcurrentStreams {
```

Comment:
This doesn't accept a stream, nor really use streams internally, which makes this a weird name + library to put it in.